client/src/presenters/CoursePresenter.tsx

```diff
@@ -3,138 +3,156 @@ import { Course } from "../models/common";
 import { useEffect, useState } from "react";
 import SectionPresenter from "./SectionPresenter";
 import {
-  ContentContainer,
-  YellowPurpleGradiant,
-  sectionHeaderStyle,
+	ContentContainer,
+	YellowPurpleGradiant,
+	sectionHeaderStyle,
 } from "../constants/styleNames";
 import { Navbar } from "../shared/Navbar";
 
 export const CoursePresenter = () => {
-  const location = useLocation();
-  const { courseCode } = useParams();
-  const [course, setCourse] = useState<Course>();
-  const [isCollapsed, setIsCollapsed] = useState<boolean>(false);
+	const location = useLocation();
+	const { courseCode } = useParams();
+	const [course, setCourse] = useState<Course>();
+	const [isCollapsed, setIsCollapsed] = useState<boolean>(false);
+	const [loading, setLoading] = useState(true);
 
-  useEffect(() => {
-    if (courseCode != null) {
-      // we should expect the url to be in the format /[CourseName][CourseNumber], where the two are split via the first numeric character's appearance
-      var courseFaculty = "";
-      var courseNumber = "";
+	useEffect(() => {
```

Comment:
could you set isLoading to true whenever we enter the useEffect... otherwise if the user searches for a course while on the coursePresenter screen, they wont get the loading screen

Reply:
and set isloading to false after the request completes